src/server/scripts/Northrend/zone_sholazar_basin.cpp

```diff
@@ -50,109 +51,147 @@ enum EngineerHelice
-                    me->SummonGameObject(184743, 5918.33f, 5372.91f, -98.770f, 0, QuaternionData(), 3s, GO_SUMMON_TIMED_DESPAWN);
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH);
-                    break;
-                case 2:
-                    Talk(SAY_WP_4);
-                    break;
-                case 7:
-                    Talk(SAY_WP_5);
-                    break;
-                case 8:
-                    me->CastSpell({ 5887.37f, 5379.39f, -91.289f }, SPELL_EXPLODE_CRYSTAL, true);
-                    me->SummonGameObject(184743, 5887.37f, 5379.39f, -91.289f, 0, QuaternionData(), 3s, GO_SUMMON_TIMED_DESPAWN);
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAUGH);
-                    break;
-                case 9:
-                    Talk(SAY_WP_6);
-                    break;
-                case 13:
-                    if (player)
-                    {
-                        player->GroupEventHappens(QUEST_DISASTER, me);
-                        Talk(SAY_WP_7);
-                    }
-                    break;
-            }
+            Start(false, false, player->GetGUID());
+            Talk(SAY_WP_1);
         }
+    }
+};
 
-        void Reset() override
-        {
-            Initialize();
-        }
+class spell_q12688_detonate_1 : public SpellScript
+{
+    PrepareSpellScript(spell_q12688_detonate_1);
 
-        void JustDied(Unit* /*killer*/) override
-        {
-            if (HasEscortState(STATE_ESCORT_ESCORTING))
-            {
-                if (Player* player = GetPlayerForEscort())
-                    player->FailQuest(QUEST_DISASTER);
-            }
-        }
+    static constexpr uint32 GO_FLAMES = 182071;
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_EXPLOSION });
+    }
 
-        void UpdateAI(uint32 uiDiff) override
+    void HandleDummyEffect(SpellEffIndex /* effIndex */)
+    {
+        if (Unit* target = GetHitUnit())
         {
-            EscortAI::UpdateAI(uiDiff);
+            target->CastSpell(target, SPELL_EXPLOSION);
 
-            if (HasEscortState(STATE_ESCORT_ESCORTING))
+            static const struct
             {
-                if (m_uiChatTimer <= uiDiff)
-                {
-                    m_uiChatTimer = 12000;
-                }
-                else
-                    m_uiChatTimer -= uiDiff;
-            }
+                Position Pos;
+                QuaternionData Quat;
+            } Flames[] =
+                    {
+                            {{ 5921.5337f, 5371.778f,  -96.338776f, 0.f }, { 0.f, 0.f, 0.6883545f,   0.72537446f }},
+                            {{ 5919.418f,  5366.918f,  -96.091125f, 0.f }, { 0.f, 0.f, 0.60876083f,  0.7933538f }},
+                            {{ 5920.2563f, 5372.4136f, -98.85827f,  0.f }, { 0.f, 0.f, -0.32556725f, 0.94551885f }}
```

Comment:
I'm not happy with spawning the flames like this. Investigating a cleaner way.

Reply:
You can spawn them in DB with negative respawn time and respawn in script